### ISSUE TITLE
Tranform _id Field to id

### ISSUE THREAD

**Issue by brianhyder** [pencilblue/pencilblue#981]
The ID field of _id is ugly in my opinion and will make things more difficult as the platform attempts to support multiple NoSQL solutions.  

A transform should be added to the DAO class to modify each result coming from MongoDB to inject an "id" field into the object.  This will make API and UI development easier.

**Comment by blakecallens** [comment]
This feels like overkill to me. Take a look at the uidService that I put in
the admin-update branch. I think there's a middle ground that doesn't
require duplicate values in each Mongo document.

**Comment by brianhyder** [comment]
I don't think it is overkill and here is why.  MongoDB is the only DB that restricts the primary identifier to a specific field name.  This means that in order to support other DBs we need to conform to this convention across our entire code base.  There are many places where we have taken steps to reduce the reference to ```someObj._id``` and changed it to ```someObj[DAO.getIdField()]``` but this is verbose and a bit ugly.  If we can bring consistency to the back-end by using a common "id" field then we gain readability, performance, and a minimal change set to support the next DB.

I took a look at the uidService that you mentioned.  This is a great intermediary step to handling the ```_id``` step on the front end while this phased approach rolls out.  However, simply handling it on the front end doesn't make our APIs consumable by those that wish to integrate with PB.  

So what I did a poor job of explaining in the issue is what this layout looks like:

1. Add a transform to the current DAO implementation to persist and inject the "id" field
2. remove all back-end references to ```someObj._id```
3. remove all front-end references to ```someObj._id```
4. Update transform to remove the now duplicate ```_id``` field

This allows us to gradually ween our users off of the ```_id``` allowing us to make gradual changes over time to better fit our available bandwidth.  Additionally, it provides immediate support to start all new implementations on the pattern that we will carry forward.
